redhat-cop/openshift-applier
# Adding support for no param file for templates + v2.x fixes & clean-up

#### What does this PR do?
This PR is the first one for the `v2.x` release, and contains the following:
- Support running with templates without params files ( #10 )
- Simplify the inventory to use `action` rather than `file_action` and `template_action`
- Fix the running with both `params` and `params_from_vars` ( #31 )

#### How should this be tested?
Multiple tests needed:
1. Run with an inventory that has both files and templates (with params) set to ensure former functionality is still valid
2. Run with an inventory that has pre/post steps set to ensure these are still operational
3. Run with an inventory that has a template with all required values set with default values, but no param file specified 
4. Run with an inventory that has the `params_from_vars` set (see `test` directory for some samples)

#### Is there a relevant Issue open for this?
resolves #10 
resolves #31

#### Who would you like to review this?
cc: @redhat-cop/openshift-applier
